cmd/poller/poller.go

```diff
@@ -903,10 +903,12 @@ func (p *Poller) publishDetails() {
 		if exp.Type != "Prometheus" {
 			continue
 		}
-		if exp.LocalHttpAddr == "localhost" || exp.LocalHttpAddr == "127.0.0.1" {
+		if exp.LocalHttpAddr == "0.0.0.0" {
+			exporterIp = localIp
+		} else if exp.LocalHttpAddr != "localhost" || exp.LocalHttpAddr == "127.0.0.1" {
```

Comment:
`else if exp.LocalHttpAddr == "localhost" || exp.LocalHttpAddr == "127.0.0.1"`

Reply:
Thanks. Yes this was failing the test hence i left PR in draft. Corrected it now.